integrasjonspunkt/src/main/java/no/difi/meldingsutveksling/config/OauthRestTemplateConfig.java

```diff
@@ -0,0 +1,89 @@
+import org.springframework.security.oauth2.client.resource.OAuth2ProtectedResourceDetails;
+import org.springframework.security.oauth2.client.resource.UserApprovalRequiredException;
+import org.springframework.security.oauth2.client.resource.UserRedirectRequiredException;
+import org.springframework.security.oauth2.client.token.AccessTokenProvider;
+import org.springframework.security.oauth2.client.token.AccessTokenRequest;
+import org.springframework.security.oauth2.client.token.DefaultAccessTokenRequest;
+import org.springframework.security.oauth2.common.DefaultOAuth2AccessToken;
+import org.springframework.security.oauth2.common.OAuth2AccessToken;
+import org.springframework.security.oauth2.common.OAuth2RefreshToken;
+import org.springframework.security.oauth2.config.annotation.web.configuration.EnableOAuth2Client;
+import org.springframework.web.client.RestOperations;
+import org.springframework.web.client.RestTemplate;
+
+import java.net.URISyntaxException;
+import java.time.Instant;
+import java.util.Date;
+
+@Configuration
+@EnableOAuth2Client
+public class OauthRestTemplateConfig {
+
+    private IntegrasjonspunktProperties props;
+    private OidcTokenClient oidcTokenClient;
+
+    @Autowired
+    public OauthRestTemplateConfig(IntegrasjonspunktProperties props, OidcTokenClient oidcTokenClient) {
+        this.props = props;
+        this.oidcTokenClient = oidcTokenClient;
+    }
+
+    @Bean
+    public RestOperations restTemplate() throws URISyntaxException {
+        if (props.getOidc().isEnable()) {
+            DefaultAccessTokenRequest atr = new DefaultAccessTokenRequest();
+            BaseOAuth2ProtectedResourceDetails resource = new BaseOAuth2ProtectedResourceDetails();
+            resource.setAccessTokenUri(String.valueOf(props.getOidc().getUrl().toURI()));
+            resource.setScope(props.getOidc().getScopes());
+            resource.setClientId(props.getOidc().getClientId());
+
+            OAuth2RestTemplate rt = new OAuth2RestTemplate(resource, new DefaultOAuth2ClientContext(atr));
+            rt.setAccessTokenProvider(new OidcAccessTokenProvider());
+            return rt;
+        }
+        return new RestTemplate();
+    }
+
+    public class OidcAccessTokenProvider implements AccessTokenProvider {
+
+        private DefaultOAuth2AccessToken getAccessToken() {
+            IdportenOidcTokenResponse oidcTokenResponse = oidcTokenClient.fetchToken();
+            DefaultOAuth2AccessToken oa2at = new DefaultOAuth2AccessToken(oidcTokenResponse.getAccessToken());
+            oa2at.setExpiration(Date.from(Instant.now().plusSeconds(250)));
+            oa2at.setScope(Sets.newHashSet(oidcTokenResponse.getScope()));
+            return oa2at;
+        }
+
+        @Override
+        public OAuth2AccessToken obtainAccessToken(OAuth2ProtectedResourceDetails details, AccessTokenRequest parameters) throws UserRedirectRequiredException, UserApprovalRequiredException, AccessDeniedException {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove the declaration of thrown exception 'org.springframework.security.oauth2.client.resource.UserRedirectRequiredException' which is a runtime exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove the declaration of thrown exception 'org.springframework.security.oauth2.client.resource.UserApprovalRequiredException' which is a runtime exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove the declaration of thrown exception 'org.springframework.security.access.AccessDeniedException' which is a runtime exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)